lib/parse.js

```diff
@@ -536,6 +538,12 @@ function tokenizer($TEXT, filename, html5_comments, shebang) {
         if (RE_OCT_NUMBER.test(num) && next_token.has_directive("use strict")) {
             parse_error("Legacy octal literals are not allowed in strict mode");
         }
+        if (numeric_separator) {
+            if(num.endsWith("_") || num.includes("__")) {
+                parse_error("Invalid numeric separator(s)");
```

Comment:
I would like to see two different syntax errors for both cases:

for multiple underscores: "Only one underscore is allowed as numeric separator"
for underscore at the en: "Numeric separators are not allowed at the end of numeric literals"